generated/nidcpower/nidcpower/lcr_measurement.py

```diff
@@ -0,0 +1,305 @@
+                self.ac_voltage = complex(data.ac_voltage_real, data.ac_voltage_imaginary)
+                self.ac_current = complex(data.ac_current_real, data.ac_current_imaginary)
+                self.z = complex(data.z_real, data.z_imaginary)
+                self.series_lcr = LCRMeasurement.LCR(
+                    inductance=data.ls, capacitance=data.cs, resistance=data.rs
+                )
+                self.parallel_lcr = LCRMeasurement.LCR(
+                    inductance=data.lp, capacitance=data.cp, resistance=data.rp
+                )
+                self.d = data.d
+                self.measurement_mode = enums.InstrumentMode(data.measurement_mode)
+                self.in_compliances = LCRMeasurement.InCompliances(
+                    dc=bool(data.dc_in_compliance), ac=bool(data.ac_in_compliance)
+                )
+                self.unbalanced = bool(data.unbalanced)
+            else:
+                for field_name, _ in LCRMeasurement._lcr_measurement_field_metadata:
+                    setattr(
+                        self,
+                        field_name,
+                        enums.InstrumentMode(data.measurement_mode)
+                        if field_name == "measurement_mode"
+                        else getattr(data, field_name),
+                    )
+        else:
+            for field_name, _ in LCRMeasurement._lcr_measurement_field_metadata:
+                setattr(
+                    self,
+                    field_name,
+                    enums.InstrumentMode(measurement_mode)
+                    if field_name == "measurement_mode"
+                    else locals()[field_name],
+                )
+
+    def get_z_magnitude_and_phase(self):
+        """get_z_magnitude_and_phase
+
+        Returns a tuple of (z_magnitude, z_phase) of this LCRMeasurement object.
+
+        Returns:
+            z_magnitude_and_phase (tuple of (float, float)):
+
+                - **z_magnitude** (float): The magnitude of the complex impedance, in ohms.
+                - **z_phase** (float): The impedance phase angle, in degrees.
+
+        """
+        return cmath.polar(self.z)
+
+    def get_y(self):
+        """get_y
+
+        Returns the admittance of this LCRMeasurement object.
+
+        Returns:
+            y (complex): The complex admittance.
+
+        """
+        if self.z == 0:
+            return complex(cmath.nan, cmath.nan)
+        return 1.0 / self.z
+
+    def get_y_magnitude_and_phase(self):
+        """get_y_magnitude_and_phase
+
+        Returns a tuple of (y_magnitude, y_phase) of this LCRMeasurement object.
+
+        Returns:
+            y_magnitude_and_phase (tuple of (float, float)):
+
+                - **y_magnitude** (float): The magnitude of the complex admittance, in siemens.
+                - **y_phase** (float): The admittance phase angle, in degrees.
+
+        """
+        return cmath.polar(self.get_y())
+
+    def get_q(self):
```

Comment:
I have renamed these getters by removing the "get_" prefix and added `@property` decorator to them so that they behave like read-only members of this class in my latest commit.